core/src/main/java/io/questdb/griffin/engine/functions/str/LPadFunctionFactory.java

```diff
@@ -0,0 +1,116 @@
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.str;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.TableUtils;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.BinaryFunction;
+import io.questdb.griffin.engine.functions.StrFunction;
+import io.questdb.std.IntList;
+import io.questdb.std.Numbers;
+import io.questdb.std.ObjList;
+import io.questdb.std.str.StringSink;
+import org.jetbrains.annotations.Nullable;
+
+public class LPadFunctionFactory implements FunctionFactory {
+    @Override
+    public String getSignature() {
+        return "lpad(SI)";
+    }
+
+    @Override
+    public Function newInstance(int position, ObjList<Function> args, IntList argPositions,
+            CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) throws SqlException {
+        final Function strFunc = args.getQuick(0);
+        final Function lenFunc = args.getQuick(1);
+        return new LPadFunc(strFunc, lenFunc);
+    }
+
+    public static class LPadFunc extends StrFunction implements BinaryFunction {
+        private final Function strFunc;
+        private final Function lenFunc;
+
+        private final StringSink sink = new StringSink();
+        private final StringSink sinkB = new StringSink();
+
+        public LPadFunc(Function strFunc, Function lenFunc) {
+            this.strFunc = strFunc;
+            this.lenFunc = lenFunc;
+        }
+
+        @Override
+        public Function getLeft() {
+            return strFunc;
+        }
+
+        @Override
+        public Function getRight() {
+            return lenFunc;
+        }
+
+        @Override
+        public int getStrLen(Record rec) {
+            final CharSequence str = strFunc.getStr(rec);
+            final int len = lenFunc.getInt(rec);
+            if (str != null && len != Numbers.INT_NaN && len >= 0) {
+                return len;
```

Comment:
The returned value will be incorrect in case `len > str.length()`.

Reply:
My tests for the same are passing.

Reply:
Not sure if I understand the point. This method returns incorrect value under certain conditions. If tests pass, it means that they don't verify this method or that they're incorrect. Am I missing something?

Reply:
In the case `len > str.length()`, the string will be padded to reach a length of `len`. So shouldn't `getStrLen()` return `len` itself?

Reply:
Yes, you're right. Sorry for the confusion, I forgot about the added padding.